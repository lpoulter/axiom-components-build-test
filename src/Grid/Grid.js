import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';
import './Grid.css';

export default class Grid extends Component {
  static propTypes = {
    /** <GridCells>. */
    children: PropTypes.node,
    /** Class name to be appended to the element */
    className: PropTypes.string,
    /*
    * Controls the direction of the grid cells
    */
    direction: PropTypes.oneOf(['row', 'column']),
    /**
     * Applies fill styling for all GridCell children.
     * See GridCell for fill explanation
     */
    fill: PropTypes.oneOf([true, 'small', 'medium', 'large']),
    /**
     * Applies fit styling for all GridCell children.
     * See GridCell for fit explanation
     */
    fit: PropTypes.oneOf([true, 'small', 'medium', 'large']),
    /**
     * Applies full styling for all GridCell children.
     * See GridCell for full explanation
     */
    full: PropTypes.oneOf([true, 'small', 'medium', 'large']),
    /** Controls the horizontal and vertical spacing between cells */
    gutters: PropTypes.oneOf([false, 'tiny', 'small', 'medium', 'large']),
    /** Controls the horizontal alignment of all cells */
    horizontalAlign: PropTypes.oneOf(['start', 'middle', 'end', 'around', 'between']),
    /** Controls the horizontal spacing between cells */
    horizontalGutters: PropTypes.oneOf([false, 'tiny', 'small', 'medium', 'large']),
    /**
     * Applies none styling for all GridCell children.
     * See GridCell for none explanation
     */
    none: PropTypes.oneOf([true, 'small', 'medium', 'large']),
    /**
     * Default behaviour of cells is to resize to 100% of the grids space when
     * on smaller screen sizes. This allows this behaviour to be suppressed.
     */
    responsive: PropTypes.bool,
    /**
     * Applies shrink styling for all GridCell children.
     * See GridCell for shrink explanation
     */
    shrink: PropTypes.oneOf([true, 'small', 'medium', 'large']),
    /** Controls the vertical alignment of all cells */
    verticalAlign: PropTypes.oneOf(['start', 'middle', 'end']),
    /** Controls the vertical spacing between cells */
    verticalGutters: PropTypes.oneOf([false, 'tiny', 'small', 'medium', 'large']),
    /** Allows cells to drop to a new line when they require more space */
    wrap: PropTypes.bool,
  };

  static defaultProps = {
    direction: 'row',
    responsive: true,
    gutters: 'medium',
    wrap: true,
  };

  render() {
    const {
      children,
      className,
      direction,
      fill,
      fit,
      full,
      gutters,
      horizontalAlign,
      horizontalGutters = gutters,
      none,
      responsive,
      shrink,
      verticalAlign,
      verticalGutters = gutters,
      wrap,
      ...rest
    } = this.props;

    const classes = classnames('ax-grid', {
      'ax-grid--responsive': responsive,
      'ax-grid--columns': direction === 'column',
      [`ax-grid--gutters-horizontal--${horizontalGutters}`]: typeof horizontalGutters === 'string',
      [`ax-grid--gutters-vertical--${verticalGutters}`]: typeof verticalGutters === 'string',
      'ax-grid--fill': fill === true,
      [`ax-grid--fill--${fill}`]: fill && fill !== true,
      'ax-grid--fit': fit === true,
      [`ax-grid--fit--${fit}`]: fit && fit !== true,
      'ax-grid--full': full === true,
      [`ax-grid--full--${full}`]: full && full !== true,
      'ax-grid--none': none === true,
      [`ax-grid--none--${none}`]: none && none !== true,
      'ax-grid--shrink': shrink === true,
      [`ax-grid--shrink--${shrink}`]: shrink && shrink !== true,
      [`ax-grid--vertical-${verticalAlign}`]: verticalAlign,
      [`ax-grid--horizontal-${horizontalAlign}`]: horizontalAlign,
      'ax-grid--wrap': wrap === true,
    }, className);

    return (
      <Base space="x6" { ...rest } className="ax-grid__container">
        <div className={ classes }>
          { children }
        </div>
      </Base>
    );
  }
}
