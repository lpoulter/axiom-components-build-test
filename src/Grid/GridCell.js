import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';

export default class GridCell extends Component {
  static propTypes = {
    children: PropTypes.node,
    /** Class name to be appended to the element */
    className: PropTypes.string,
    /*
    * Controls the direction of the grid cells
    */
    direction: PropTypes.oneOf(['row', 'column']),
    /**
     * Sizes itself according to the size of its contents. This causes the grid
     * to attempt to evenly distribute the cells, but allows cells to increase
     * their width to provide more room for the contents.
     *
     * When a filled cell exceeds the available room (relative to siblings) it
     * will drop onto its own line. Combined with Grids wrapping property, this
     * can controlled not to shrink siblings down.
     *
     * This can be configured to be all of the time or at specific breakpoints.
     */
    fill: PropTypes.oneOf([true, 'small', 'medium', 'large']),
    /**
     * Sizes itself irrespective of its contents. This causes the grid to evenly
     * distribute all children cells. If the grid has wrapping enabled, the cell
     * may drop to a new line when there is no room left.
     *
     * This can be configured to be all of the time or at specific breakpoints.
     */
    fit: PropTypes.oneOf([true, 'small', 'medium', 'large']),
    /**
     * Sizes itself to 100% of the available width within a grid. This causes
     * all other siblings to placed vertically of itself.
     *
     * This can be configured to be all of the time or at specific breakpoints.
     */
    full: PropTypes.oneOf([true, 'small', 'medium', 'large']),
    /**
     * Sizes itself according to the size of its contents, similar to `shrink` it
     * will no increase to any available space but also will not decrease below
     * it's own content size.
     *
     * This can be configured to be all of the time or at specific breakpoints.
     */
    none: PropTypes.oneOf([true, 'small', 'medium', 'large']),
    /**
     * Sizes itself according to the size of its contents, but does not increase
     * to any available width within a grid. This allows all other siblings to
     * consume the available space.
     *
     * This can be configured to be all of the time or at specific breakpoints.
     */
    shrink: PropTypes.oneOf([true, 'small', 'medium', 'large']),
    /**
     * Allows a grid cell function as a flex container itself.
     */
    subGrid: PropTypes.bool,
    /**
     * Vertically aligns itself relative to other cells within a grid.
     */
    verticalAlign: PropTypes.oneOf(['start', 'middle', 'end']),
    /** Percentage width */
    width: PropTypes.number,
  };

  render() {
    const {
      children,
      className,
      direction,
      fill,
      fit,
      full,
      none,
      width,
      shrink,
      subGrid,
      verticalAlign,
      ...rest
    } = this.props;

    const classes = classnames('ax-grid__cell', {
      [`ax-grid__cell--${verticalAlign}`]: verticalAlign,
      'ax-grid__cell--fill': fill === true,
      'ax-grid__cell--fit': fit === true,
      'ax-grid__cell--full': full === true,
      'ax-grid__cell--none': none === true || width,
      'ax-grid__cell--shrink': shrink === true,
      [`ax-grid__cell--fit--${fit}`]: fit && fit !== true,
      [`ax-grid__cell--fill--${fill}`]: fill && fill !== true,
      [`ax-grid__cell--full--${full}`]: full && full !== true,
      [`ax-grid__cell--none--${none}`]: none && none !== true,
      [`ax-grid__cell--shrink--${shrink}`]: shrink && shrink !== true,
      'ax-grid__cell--sub-grid': subGrid,
      'ax-grid--columns': subGrid && direction === 'column',
    }, className);

    const styles = {
      width: width && `${Math.max(0, Math.min(width, 100))}%`,
    };

    return (
      <Base { ...rest } className={ classes } style={ styles }>
        { children }
      </Base>
    );
  }
}
