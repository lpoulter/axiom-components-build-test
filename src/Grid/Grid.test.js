import React from 'react';
import renderer from 'react-test-renderer';
import Grid from './Grid';

const getComponent = (props = {}) =>
  renderer.create(
    <Grid { ...props }>
      <div></div>
    </Grid>
  );

describe('Grid', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with className', () => {
    const component = getComponent({ className: 'foo' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with responsive false', () => {
    const component = getComponent({ responsive: false });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with fill', () => {
    [true, 'small', 'medium', 'large'].forEach((fill) => {
      it(String(fill), () => {
        const component = getComponent({ fill });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with fit', () => {
    [true, 'small', 'medium', 'large'].forEach((fit) => {
      it(String(fit), () => {
        const component = getComponent({ fit });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with full', () => {
    [true, 'small', 'medium', 'large'].forEach((full) => {
      it(String(full), () => {
        const component = getComponent({ full });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with gutters', () => {
    [false, 'tiny', 'small', 'medium', 'large'].forEach((gutters) => {
      it(String(gutters), () => {
        const component = getComponent({ gutters });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with horizontalAlign', () => {
    ['start', 'middle', 'end', 'around', 'between'].forEach((horizontalAlign) => {
      it(horizontalAlign, () => {
        const component = getComponent({ horizontalAlign });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with horizontalGutters', () => {
    [false, 'tiny', 'small', 'medium', 'large'].forEach((horizontalGutters) => {
      it(String(horizontalGutters), () => {
        const component = getComponent({ horizontalGutters });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with shrink', () => {
    [true, 'small', 'medium', 'large'].forEach((shrink) => {
      it(String(shrink), () => {
        const component = getComponent({ shrink });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with verticalAlign', () => {
    ['start', 'middle', 'end'].forEach((verticalAlign) => {
      it(verticalAlign, () => {
        const component = getComponent({ verticalAlign });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with verticalGutters', () => {
    [false, 'tiny', 'small', 'medium', 'large'].forEach((verticalGutters) => {
      it(String(verticalGutters), () => {
        const component = getComponent({ verticalGutters });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
