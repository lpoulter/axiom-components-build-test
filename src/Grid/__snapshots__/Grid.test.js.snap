// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders with className 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--wrap foo"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with defaultProps 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with fill large 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--fill--large ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with fill medium 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--fill--medium ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with fill small 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--fill--small ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with fill true 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--fill ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with fit large 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--fit--large ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with fit medium 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--fit--medium ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with fit small 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--fit--small ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with fit true 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--fit ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with full large 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--full--large ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with full medium 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--full--medium ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with full small 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--full--small ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with full true 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--full ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with gutters false 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with gutters large 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--large ax-grid--gutters-vertical--large ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with gutters medium 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with gutters small 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--small ax-grid--gutters-vertical--small ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with gutters tiny 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--tiny ax-grid--gutters-vertical--tiny ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with horizontalAlign around 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--horizontal-around ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with horizontalAlign between 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--horizontal-between ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with horizontalAlign end 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--horizontal-end ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with horizontalAlign middle 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--horizontal-middle ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with horizontalAlign start 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--horizontal-start ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with horizontalGutters false 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-vertical--medium ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with horizontalGutters large 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--large ax-grid--gutters-vertical--medium ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with horizontalGutters medium 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with horizontalGutters small 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--small ax-grid--gutters-vertical--medium ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with horizontalGutters tiny 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--tiny ax-grid--gutters-vertical--medium ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with responsive false 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with shrink large 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--shrink--large ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with shrink medium 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--shrink--medium ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with shrink small 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--shrink--small ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with shrink true 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--shrink ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with verticalAlign end 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--vertical-end ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with verticalAlign middle 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--vertical-middle ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with verticalAlign start 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--vertical-start ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with verticalGutters false 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with verticalGutters large 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--large ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with verticalGutters medium 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with verticalGutters small 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--small ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;

exports[`Grid renders with verticalGutters tiny 1`] = `
<div
  className="ax-grid__container ax-space--x6"
>
  <div
    className="ax-grid ax-grid--responsive ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--tiny ax-grid--wrap"
  >
    <div />
  </div>
</div>
`;
